### DISCUSSION TITLE
Q-table re-rendering on row select?

### DISCUSSION THREAD

**Discussion by kouellette** [quasarframework/quasar#10988] [General - Components / Directives / etc] [answered]
Hey All,

I'm brand new to quasar and relatively new to vue so I apologize in advance if this is just a blatant misuse of the component or vue itself. Regardless, I'm happy to be here!

I'm building a data table using the body slot in which one of the columns uses some conditional logic based on the value of the cell to set the color of a badge. There are a handful of conditions that need to be checked so I've created a helper function which I bind to the badge's color attribute (i.e. `:color="myFunction(slotProps.row.value)`). This works as expected and if I update the underlying data, reactivity kicks in, re-evaluates `myFunction` for the single cell, and updates the badge color. However, what I've noticed is when selecting or de-selecting a row (`selection="multiple"` in this case), `myFunction` is re-evaluated for every single row in the table. When dealing with a table that is displaying a few hundred rows, this induces some lag into the row's checkbox being checked.

Is there something inherently wrong with this approach or how I'm trying to implement it? Any insight and/or suggestions would be greatly appreciated!

[Here's a codepen](https://codepen.io/kouellette/pen/KKvdogz) that closely mimics my current applications and shows the helper function being executed for each table row when a row is selected or de-selected.

**Comment by metalsadman** [comment, marked as answer]
it's just how vue reacts to method bindings, you can see it in this test https://codepen.io/metalsadman/pen/OJjyBQo?editors=1010
change the input, see change in color of the test text.
also see https://forum.vuejs.org/t/when-does-vue-reevaluate-a-method-binding/19516

In your case you can use a computed data for your rows and have an evaluated color property added, so it will only re-evaluates when the dependency itself changes. see https://codepen.io/metalsadman/pen/GRvpYjy?editors=1010

**Reply by kouellette** [reply]
Thanks for the insight and the codepen! This certainly gets me closer to what I'm looking for and is much more performant. Based on my understanding of the vue reactivity system, I'm pretty confident the answer to this question is no, but I'll go ahead and ask anyway just to confirm. Is there a way for the reactivity to only re-render the single row that had data changed? For example, in my application I have hundreds of rows, and data for rows will update fairly regularly. With this current approach, every time the data for a row is updated, the entire table will be re-rendered since it's bound to the array of objects. It would be a nice performance boost if this could be implemented in a way that would only re-render the given row associated with the object that was updated.

As I said, I believe the answer is no, this isn't possible, but I wanted to confirm that understanding. I also know this is related to vue and not quasar so I apologize if this is getting off topic.

~ Kyle
